bolt-modules/out/lib/puppet/functions/out/message.rb

```diff
@@ -22,55 +24,11 @@ def output_message(message)
         .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'out::message')
     end
 
-    executor = Puppet.lookup(:bolt_executor)
-    # Send Analytics Report
-    executor.report_function_call(self.class.name)
-
-    executor.publish_event(type: :message, message: stringify(message))
-
-    nil
-  end
-
-  def stringify(message)
-    formatted = format_message(message)
-    if formatted.is_a?(Hash) || formatted.is_a?(Array)
-      ::JSON.pretty_generate(formatted)
-    else
-      formatted
-    end
-  end
-
-  def format_message(message)
-    case message
-    when Array
-      message.map { |item| format_message(item) }
-    when Bolt::ApplyResult
-      format_apply_result(message)
-    when Bolt::Result, Bolt::ResultSet
-      # This is equivalent to to_s, but formattable
-      message.to_data
-    when Bolt::RunFailure
-      formatted_resultset = message.result_set.to_data
-      message.to_h.merge('result_set' => formatted_resultset)
-    when Hash
-      message.each_with_object({}) do |(k, v), h|
-        h[format_message(k)] = format_message(v)
-      end
-    when Integer, Float, NilClass
-      message
-    else
-      message.to_s
+    Puppet.lookup(:bolt_executor).tap do |executor|
+      executor.report_function_call(self.class.name)
+      executor.publish_event(type: :message, message: Bolt::Util::Format.stringify(message))
     end
-  end
 
-  def format_apply_result(result)
-    logs = result.resource_logs&.map do |log|
-      # Omit low-level info/debug messages
-      next if %w[info debug].include?(log['level'])
-      indent(2, format_log(log))
-    end
-    hash = result.to_data
-    hash['logs'] = logs unless logs.empty?
-    hash
+    nil
```

Comment:
Is `nil` serving as a default return value? Not sure why `hash` and `nil` are at the end of the function.

Reply:
Yep. The last evaluated statement in a method is its return value. We specifically don't want the `out::*` and `log::*` functions to return anything, so we return `nil` here.